src/physics/em/detail/SeltzerBergerInteractor.i.hh

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Construct with shared and state data.
+ *
+ * The incident gamma energy must be at least twice the electron rest mass.
+ */
+SeltzerBergerInteractor::SeltzerBergerInteractor(
+    const SeltzerBergerDeviceRef& device_pointers,
+    const ParticleTrackView&      particle,
+    const Real3&                  inc_direction,
+    const CutoffView&             cutoffs,
+    StackAllocator<Secondary>&    allocate,
+    const MaterialView&           material)
+    : device_pointers_(device_pointers)
+    , particle_id_(particle.particle_id())
+    , inc_energy_(particle.energy())
+    , inc_momentum_(particle.momentum())
+    , inc_direction_(inc_direction)
+    , cutoffs_(cutoffs)
+    , allocate_(allocate)
+    , material_(material)
+{
+    CELER_EXPECT(particle_id_ == device_pointers_.ids.electron
+                 || particle_id_ == device_pointers_.ids.positron);
+    // Check energy range is applicable for this model
+    energy_val_min_
+        = celeritas::min(cutoffs_.energy(device_pointers_.ids.gamma).value(),
+                         inc_energy_.value());
+    CELER_ENSURE(energy_val_min_ < inc_energy_.value());
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Pair-production using the Seltzer-Berger model.
+ *
+ * See section 10.2.1 of the Geant physics reference 10.6.
+ */
+template<class Engine>
+CELER_FUNCTION Interaction SeltzerBergerInteractor::operator()(Engine& rng)
+{
+    // Allocate space for the bremss photon
+    Secondary* gamma_secondary = this->allocate_(1);
+    if (gamma_secondary == nullptr)
+    {
+        // Failed to allocate space for the secondary
+        return Interaction::from_failure();
+    }
+
+    // Density correction
+    const auto migdal = 4.0 * constants::pi * constants::r_electron
+                        * ipow<2>(constants::lambdabar_electron);
+    real_type density_factor   = material_.electron_density() * migdal;
+    real_type total_energy_val = inc_energy_.value()
+                                 + device_pointers_.electron_mass;
+    real_type density_correction = density_factor * ipow<2>(total_energy_val);
+
+    // Outgoing photon energy sampler
+    SBEnergyDistribution sample_gamma_energy{
+        device_pointers_,
+        inc_energy_,
+        ElementId{0},
+        EnergySq{density_correction},
+        cutoffs_.energy(device_pointers_.ids.gamma)};
+    Energy gamma_exit_energy = sample_gamma_energy(rng);
+
+    // Cross-section scaling for positrons
+    if (particle_id_ == device_pointers_.ids.positron)
+    {
+        SBPositronXsCorrector scale_xs(
+            units::MevMass{device_pointers_.electron_mass},
+            material_.element_view(ElementComponentId{0}),
+            cutoffs_.energy(device_pointers_.ids.gamma),
+            inc_energy_);
+        gamma_exit_energy
+            = Energy{gamma_exit_energy.value() * scale_xs(gamma_exit_energy)};
```

Comment:
This block isn't quite right (although we'll need to use the constructor you've enumerated). The `XsCorrector` needs to be used as part of the energy distribution, which I didn't implement because I didn't want to interfere with your ongoing work on the SB interactor 😬 . The XsCorrector should modify the energy distribution to be sampled from, rather than scaling the exiting energy as written here.